mlir/include/mlir/Reducer/ReductionPatternInterface.h

```diff
@@ -51,6 +52,31 @@ class DialectReductionPatternInterface
   DialectReductionPatternInterface(Dialect *dialect) : Base(dialect) {}
 };
 
+/// This interface extends the `dialectreductionpatterninterface` by allowing
```

Comment:
Could we add this as a option to the above interface and have it be empty by default?

The easiest may even be to convert these plain C++ ones to using ODS (move definition to TableGen file) then we'd get extra documentation generation too and specifying default implementations is supported there.

That way, there is both no breakage but also just one interface. And it feels like these are peers. One would ideally want most to be the former (as then search is handled by infra, while the tester one is more internalized state/search).